### DISCUSSION TITLE
How to disable healthchecks?

### DISCUSSION THREAD

**Discussion by gkibria** [coollabsio/coolify#2140] [Q&A]
Hi, I am not getting any healthchecks option in the ui? I tried several docker compose projects like (n8n, filebrowser etc.) but every service has healthchecks issue and not working (404).

There is a mention in the docs of disabling healthchecks in the UI but there is no such options in my instance.

I am using 274 version.

![Screenshot 2024-05-03 at 19 00 33](https://github.com/coollabsio/coolify/assets/7963363/8c92858a-be62-41fd-8d03-52d235d42f89)

Any idea? Thanks.

**Comment by andrasbacsai** [comment]
Compose files does not have the same healthcheck options as other applications. 

You need to modify the healthcheck part of the compose file.

You can compare your n8n to this: https://github.com/coollabsio/coolify/blob/main/templates/compose/n8n.yaml

**Comment by JuliaBonita** [comment]
I don't see any healthcheck options either. I've deployed several apps, including popular ones like Postregsql and static websites, and I have never seen any healthcheck options for any of the apps or services. 

If we're using third-party apps that are automatically deployed in Coolify with their pre-defined compose files, how do we enable and disable the healthchecks in the UI?

The documentation doesn't say anything about this. It just assumes that the Healthchecks feature is visible in the Coolify UI, but it never is for any of the apps I've deployed so far.

**Comment by Blockchain-Oracle** [comment]
## Fix for Docker Healthcheck Issues

If you're experiencing healthcheck failures in Docker or Coolify deployments, ensure you have the correct setup:

### 1. Install wget in your Dockerfile

Make sure your Dockerfile installs wget in the runtime stage:

```dockerfile
# Install runtime dependencies including wget for healthchecks
RUN apt-get update && apt-get install -y \
    wget \
    # other dependencies...
    && rm -rf /var/lib/apt/lists/*
```

### 2. Configure healthcheck in docker-compose.yml

Use the CMD-SHELL format with wget instead of curl (I tried curl and it consistently failed):

```yaml
healthcheck:
  test: ["CMD-SHELL", "wget -qO- http://127.0.0.1:${PORT:-3001}/api/health"]
  interval: 5s
  timeout: 20s
  retries: 10
```

**Important notes:**
- I specifically tested with curl and it didn't work in my environment
- wget proved more reliable in minimal container environments
- Replace `/api/health` with your actual health endpoint path / valid endpoint
- The `${PORT:-3001}` syntax provides a fallback port if not defined in env
- This format is proven to work with Coolify and other Docker orchestration tools

This solution fixed our deployment issues after standard healthcheck configurations kept failing.

**Reply by tapsuns5** [reply]
Man, this was great. Kept running into health check issues, and this resolved!
